### PR TITLE
Consider boxing when checking whether value matches type

### PR DESCRIPTION
**Pull request by findepi**
This fixes index join in TPCH connector (`IndexedTpchConnectorFactory`)

### CODE REVIEW

**Comment by findepi** [comment]
cc @nezihyigitbasi

**Comment by findepi** [comment]
This is unrelated to https://github.com/prestodb/presto/issues/9800 -- the code was failing _before_ and _after_ https://github.com/prestodb/presto/pull/9701

File: presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java [hunk 1]
```diff
@@ -536,7 +537,7 @@ protected Node visitShowCreate(ShowCreate node, Void context)
                     }
 
                     PropertyMetadata<?> property = allTableProperties.get(propertyName);
-                    if (!property.getJavaType().isInstance(value)) {
+                    if (!Primitives.wrap(property.getJavaType()).isInstance(value)) {
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java]: Is it possible to test it?

Comment (findepi) [reply presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java]: I tried. `blackhole` has numeric table properties... but it doesn't report them in `show create table..`. Apparently similar thing for Hive. So no.

**Pull request merged**